cmd/archival/main.go

```diff
@@ -0,0 +1,162 @@
+	stakingKeyFile = flag.String("staking_key", "./.stakingkey", "the private key file of the harmony node")
+	// Key file to store the private key
+	keyFile = flag.String("key", "./.hmykey", "the private key file of the harmony node")
+	// isBeacon indicates this node is a beacon chain node
+	isBeacon = flag.Bool("is_beacon", false, "true means this node is a beacon chain node")
+	// isNewNode indicates this node is a new node
+	isNewNode    = flag.Bool("is_newnode", false, "true means this node is a new node")
+	accountIndex = flag.Int("account_index", 0, "the index of the staking account to use")
+	// isLeader indicates this node is a beacon chain leader node during the bootstrap process
+	isLeader = flag.Bool("is_leader", false, "true means this node is a beacon chain leader node")
+	// logConn logs incoming/outgoing connections
+	logConn = flag.Bool("log_conn", false, "log incoming/outgoing connections")
+)
+
+func printVersion(me string) {
+	fmt.Fprintf(os.Stderr, "Harmony (C) 2018. %v, version %v-%v (%v %v)\n", path.Base(me), version, commit, builtBy, builtAt)
+	os.Exit(0)
+}
+
+// InitLDBDatabase initializes a LDBDatabase. isBeacon=true will return the beacon chain database for normal shard nodes
+func InitLDBDatabase(ip string, port string, freshDB bool, isBeacon bool) (*ethdb.LDBDatabase, error) {
+	var dbFileName string
+	if isBeacon {
+		dbFileName = fmt.Sprintf("./db/harmony_beacon_%s_%s", ip, port)
+	} else {
+		dbFileName = fmt.Sprintf("./db/harmony_%s_%s", ip, port)
+	}
+	if freshDB {
+		var err = os.RemoveAll(dbFileName)
+		if err != nil {
+			fmt.Println(err.Error())
+		}
+	}
+	return ethdb.NewLDBDatabase(dbFileName, 0, 0)
+}
+
+func initSetup() {
+	if *versionFlag {
+		printVersion(os.Args[0])
+	}
+
+	// Logging setup
+	utils.SetPortAndIP(*port, *ip)
+
+	// Add GOMAXPROCS to achieve max performance.
+	runtime.GOMAXPROCS(1024)
+
+	// Set up randomization seed.
+	rand.Seed(int64(time.Now().Nanosecond()))
+
+	if len(utils.BootNodes) == 0 {
+		bootNodeAddrs, err := utils.StringsToAddrs(utils.DefaultBootNodeAddrStrings)
+		if err != nil {
+			panic(err)
+		}
+		utils.BootNodes = bootNodeAddrs
+	}
+}
+
+func createArchivalNode() (*node.Node, *nodeconfig.ConfigType) { //Mix of setUpConsensusAndNode and createGlobalConfig
+	var err error
+	nodeConfig := nodeconfig.GetGlobalConfig()
+
+	// Currently we hardcode only one shard.
+	nodeConfig.ShardIDString = "0"
+	// Consensus keys are the BLS12-381 keys used to sign consensus messages
+	nodeConfig.ConsensusPriKey, nodeConfig.ConsensusPubKey = utils.GenKey(*ip, *port)
+	// P2p private key is used for secure message transfer between p2p nodes.
+	nodeConfig.P2pPriKey, _, err = utils.LoadKeyFromFile(*keyFile)
+	if err != nil {
+		panic(err)
+	}
+
+	// Initialize leveldb for main blockchain and beacon.
+	if nodeConfig.MainDB, err = InitLDBDatabase(*ip, *port, *freshDB, false); err != nil {
+		panic(err)
+	}
+	if nodeConfig.BeaconDB, err = InitLDBDatabase(*ip, *port, *freshDB, true); err != nil {
```

Comment:
For archival node, we only need to store one DB depends on which shard the archival node belongs to

Reply:
Done